### PR TITLE
Flexible extension to RFrames

### LINKED ISSUE #331: Extension to RFrames
**Is your feature request related to a problem? Please explain.**
When specifying a relative frame range it would be would be nice to have an option to specify if the reference frame was the previous object's first or last frame. A particular use case is when objects need to be added to the canvas one after another based on their starting frame.

**Describe the solution you'd like**
Add the argument `from_start::Bool` to `RFrames` to indicate the frame range for this object. 
Example:
```
ball = Object(Rframes(5:10, true), (args...)->somefunc())
```

**Describe alternatives you've considered**
A workaround is to create a function to access the starting frame of the desired object in a way like `start(obj)`. The only problem seems to be when the previous object (or a reference to it) is not known.

**Additional context**
Both the methods were suggested by @Wikunia on Javis' zulip [thread](https://julialang.zulipchat.com/#narrow/stream/253420-javis/topic/Extension.20to.20Rframes.3F).

### PR DESCRIPTION
**Pull request by codejaeger**
## PR Checklist

If you are contributing to `Javis.jl`, please make sure you are able to check off each item on this list:

- [ ] Did I update `CHANGELOG.md` with whatever changes/features I added with this PR?
- [x] Did I make sure to only change the part of the file where I introduced a new change/feature?
- [x] Did I cover all corner cases to be close to 100% test coverage (if applicable)?
- [ ] Did I properly add Javis dependencies to the `Project.toml` + set an upper bound of the dependency (if applicable)?
- [x] Did I properly add test dependencies to the `test` directory (if applicable)?
- [x] Did I check relevant tutorials that may be affected by changes in this PR?
- [x] Did I clearly articulate why this PR was made the way it was and how it was made?

**Link to relevant issue(s)**
Closes #331 

**How did you address these issues with this PR? What methods did you use?**
As discussed in the issue there were 2 approaches to make `RFrames` more flexible. One was to supply additional flags to accommodate the different cases. Other is to let the user specify the starting reference and (optionally) the end frame. 
The latter allows a wide range of options like specifying the frame range based on a previously created object.
Any suggestions @Wikunia @TheCedarPrince @Sov-trotter ?

### CODE REVIEW

File: test/frames.jl [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    @test c.frames.frames == 5:7
+    @test d.frames.frames == 5:7
+end
+
+@testset "Check RFrame helper function for previous object" begin
+    function dummy(args...) end
+    video = Video(10, 10)
+    Background(1:20, dummy)
+    a = Object(1:3, (args...) -> O) # 1:3
+    b = Object(RFrames(2:5, prev_start(), default_last()), (args...) -> O) # 3:6
```
Comment (Wikunia) [inline test/frames.jl:21]: I find this a little hard to read. What exactly is the meaning of `default_last()` when combined with `prev_start` here?

**Comment by Wikunia** [comment]
Thanks for opening this PR @codejaeger 
I find some of the variants currently a bit hard to read. Have we thought of the following before?
Another idea might be to use `Frames(prev_start()+5, prev_start()+10)` to provide a start and end frame. Or maybe easier
`Frames(prev_start()+5, 10)` and the second parameter is always the length. I would more or less leave `RFrames` as it is then and just have a generic approach with `Frames` where the user can do whatever. Default to length of the object action seems reasonable to me but using `Frames(prev_start(); stop=100)` would make it straightforward I think.

**Comment by Wikunia** [comment]
I think a good way to have a lot of flexibility would be to use:
```julia
macro Frames(start, len)
         return :(()->$start:$len)
end
```
and then with simple functions like this:
```julia
prev_start() = CURRENT_OBJECT[1].frames.frames[1]
```
This would be possible: 
```julia
@Frames(prev_start(), 10)
```
Javis just needs to call the anonymous macro function then to get the frames.
We have to check how we get something like `prev_start` work with actions as well and have maybe an `parent_start` to get the `object` instead of the previous action as an example.

Additionally maybe we can provide a `macro` with `start, stop` instead of `start, len` if that is desired.

**Comment by Wikunia** [comment]
@codejaeger you can let me know here if you want to continue working on it. If not I might implement this myself in the next couple of days.

**Comment by codejaeger** [comment]
Hey @Wikunia , here are my views and thoughts on the approach. Your idea seems a much more readable way to do the job than what the PR proposes. I have just a few doubts on the approach.
1. Is the usage going to be similar to `Object(@Frames(prev_start(), 10), (args...) -> dummy())`?
2. The `stop` parameter seems a global frame number to me. The way frames are stored in actions is relative to the Object. So this parameter does not seem very useful for action but it is important for Objects anyways.
3. The reason I was able to adjust `parent_start` to actions was because I had the type of the elem the frame belonged to available from inside the `get_frames` method.
4. I tried this code and right now it seems the CURRENT_OBJECT array is not updated at the end of Object creation. Is that a bug? I don't see it being used anywhere so it might have gone undetected.

**Comment by Wikunia** [comment]
1. Yes it will be that syntax
2. We might want to have it relative to either the global one or the parent object. Similar to how the frames work now for the action
3. I'll need to look into this more
4. The current object only changes inside render as it should be independent of when it was created.
5. I think it should work well with actions and should just be relative to the parent object.

**Comment by Wikunia** [comment]
Can we close this @codejaeger ?

**Comment by codejaeger** [comment]
Yes, sure @Wikunia .
